src/components/needLeadViewPage/needsViewTable.tsx

```diff
@@ -0,0 +1,287 @@
+      justifyContent: "space-around",
+      alignItems: "center",
+      color: "#26396a"
+    },
+    headerText: {
+      fontFamily: "inherit",
+      fontWeight: 400,
+      fontSize: "20px"
+    },
+    inputBox: {
+      display: "flex",
+      width: "40%",
+      height: "25px",
+      justifyContent: "space-between",
+      alignItems: "center",
+      border: "1px solid #c3c3c3",
+      background: "white",
+      paddingRight: "2px",
+      marginRight: "0px"
+    },
+    input: {
+      width: "100%",
+      border: "none",
+      height: "23px",
+      outline: "none"
+    },
+    searchIcon: {
+      color: "#616161"
+    },
+    table: {
+      width: "100%",
+      minWidth: "908px",
+      minHeight: "425px"
+    },
+    innerTable: {
+      width: "100%",
+      height: "42vh",
+      minWidth: "800px",
+      minHeight: "285",
+      border: "1px solid #c3c3c3",
+      marginTop: "5px",
+      fontSize: "13px",
+      fontFamily: "sans - serif",
+      fontWeight: 400
+    },
+    header: {
+      color: "#26396a",
+      textAlign: "start",
+      fontWeight: 400,
+      transform: "scaleY(0.9)"
+    },
+    td: {
+      padding: "0px 10px 15px 5px",
+      transform: "scaleY(0.9)",
+      fontWeight: 400
+    },
+    bodyBox: {
+      color: "#77797a"
+    },
+    footer: {
+      display: "flex",
+      justifyContent: "center",
+      alignItems: "center",
+      color: "#5a5a5a"
+    },
+    arrowIcon: {
+      cursor: "pointer"
+    },
+    tableCustomBorder: {
+      border: "1px solid #86888b",
+      position: "absolute",
+      width: "100%",
+      height: "calc(100% - 35px)",
+      top: "35px",
+      minWidth: "908px"
+    }
+  }));
+  const styles = useStyles();
+
+  const { data: needsData } = useQuery(GET_NEEDS, {
+    variables: {
+      request: {
+        options: {
+          page: page,
+          limit: 10
+        },
+        filters: {}
+      }
+    },
+    fetchPolicy: "cache-and-network"
+  });
+
+  const { data: filteredNeeds } = useQuery(GET_NEEDS, {
+    variables: {
+      request: {}
+    },
+    fetchPolicy: "cache-and-network"
+  });
+
+  useEffect(() => {
+    if (filterStr !== "") {
+      let totalNeeds = filteredNeeds?.getAllNeeds?.needs || [];
+      totalNeeds = totalNeeds
+        .filter((need: any) => {
+          return need.itemName.includes(filterStr);
+        })
+        .slice(0, 10);
+      let space = 10 - totalNeeds.length;
+      for (let i = 0; i < space; i++) {
+        totalNeeds.push({
+          urgencyLevel: "",
+          itemName: "",
+          createdAt: "",
+          specifications: "",
+          quantity: "",
+          budget: "",
+          id: `spacer-need-${Math.random()}`
+        });
+      }
+      setNeeds(totalNeeds);
+      setTotalPage(1);
+    } else {
+      const totalNeeds = filteredNeeds?.getAllNeeds?.needs || [];
+      const needs = needsData?.getAllNeeds?.needs || [];
+      let space;
+      if (needs.length !== 10) {
```

Comment:
needs.lenght < 10 ? And why are you filling up the table with empty rows?

Reply:
Removed the logic for that, I was using it to correctly space the items but I was able to fix it with CSS.